deploy/6_proxy_safe.js

```diff
@@ -32,17 +36,21 @@ module.exports = async (hre) => {
     const safe_contracts = require("@gnosis.pm/safe-contracts");
     proxy_address = await safe_contracts.calculateProxyAddress(proxy_factory_contract, master_contract.address, setupData, nonce);
 
-    await proxy_factory_contract.createProxyWithNonce(master_contract.address, setupData, nonce).then((tx) => tx.wait());
+    await proxy_factory_contract.createProxyWithNonce(singleton_contract.address, setupData, nonce).then((tx) => tx.wait());
     console.log("Safe proxy deployed to", proxy_address);
 
     // Log safe owners
     const ethers = require('ethers');
     default_mnemonic = "test test test test test test test test test test test junk";
-    path_string = "m/44'/60'/0'/0/n"
+    path_string = "m/44'/60'/0'/0/n";
 
     for (let i = 10; i < 14; i++) {
-        path = path_string.replace('n', i.toString())
+        path = path_string.replace('n', i.toString());
         wallet = ethers.Wallet.fromMnemonic(default_mnemonic, path);
         console.log("Safe proxy owner", (i - 10).toString(), "address:", wallet.address, "key:", wallet.privateKey);
     }
+
+    // Verify proxy deployment
+    proxy_contract = await hre.ethers.getContractAt("GnosisSafeProxy", proxy_address);
+    console.log(proxy_contract.masterCopy());
```

Comment:
Is this correct? Andrey pointed out that we should test with `masterCopy()` and try to retrieve the safe address that way. This returns a `TypeError: proxy_contract.masterCopy is not a function` ATM. That is defined in the proxy contract as:
```
interface IProxy {
    function masterCopy() external view returns (address);
}
```

Reply:
I finished it now, was just a small fix.